src/main/java/org/jabref/gui/EntryTypeView.java

```diff
@@ -168,4 +178,82 @@ private void setEntryTypeForReturnAndClose(Optional<BibEntryType> entryType) {
+                case Unpublished:
+                    return Localization.lang("A document having an author and title, but not formally published.");
+                case BookInBook:
+                    return Localization.lang("This type is similar to inbook but intended for works originally published as astand-alone book.");
+                case InReference:
+                    return Localization.lang("An article in a work of reference.");
+                case MvBook:
+                    return Localization.lang("A multi-volume book.");
+                case MvCollection:
+                    return Localization.lang("A multi-volume collection.");
+                case MvProceedings:
+                    return Localization.lang("A multi-volume proceedings entry.");
+                case MvReference:
+                    return Localization.lang("A multi-volume reference entry. The standard styles will treat this entry typeas an alias for mvcollection.");
+                case Online:
+                    return Localization.lang("This entry type is intended for sources such as web sites which are intrinsically online resources.");
+                case Reference:
+                    return Localization.lang("A single-volume work of reference such as an encyclopedia or a dictionary.");
+                case Report:
+                    return Localization.lang("A technical report, research report, or white paper published by a university or someother institution.");
+                case Set:
+                    return Localization.lang("An entry set is a group of entries which are cited as a single reference and listed as a single item in the bibliography.");
+                case SuppBook:
+                    return Localization.lang("Supplemental material in a book. This type is provided for elements such as prefaces, introductions, forewords, afterwords, etc. which often have a generic title only.");
+                case SuppCollection:
+                    return Localization.lang("Supplemental material in a collection.");
+                case SuppPeriodical:
+                    return Localization.lang("Supplemental material in a periodical. This type may be useful when referring to items such as regular columns, obituaries, letters to the editor, etc. which only have a generic title.");
+                case Thesis:
+                    return Localization.lang("A thesis written for an educational institution to satisfy the requirements for a degree.");
+                case WWW:
+                    return Localization.lang("An alias for online, provided for jurabib compatibility.");
+                case Software:
+                    return Localization.lang("Computer software. The standard styles will treat this entry type as an alias for misc.");
+                case DATESET:
+                    return Localization.lang("A data set or a similar collection of (mostly) raw data.");
```

Comment:
After you have finished the descriptions, you will need to add them to the English localization file: `src/main/resources/l10n/JabRef_en.properties`.

Reply:
> 
> 
> @dimitra-karadima Right clicking an entry in the main table calls
> 
> https://github.com/JabRef/jabref/blob/bbf3ab475e1300715cab72620b2c7c05dcce13bf/src/main/java/org/jabref/gui/menus/ChangeEntryTypeMenu.java#L50
> 
> which further calls
> 
> https://github.com/JabRef/jabref/blob/bbf3ab475e1300715cab72620b2c7c05dcce13bf/src/main/java/org/jabref/gui/menus/ChangeEntryTypeMenu.java#L57-L58

I have found the populateComplete method but when I entered the getDescpription() and setTooltip(), no tooltip was shown. I will look into this more and try to figure it out.

Reply:
@systemoperator I have made some changes in my new commit but the re-request review button does not work for you, I don't know why. So, please take a look and share your thoughts!

Reply:
@systemoperator check my latest commit! And if you have any new thoughts on how to add those tooltips in the entry editor on in the main table it would be really helpful! I am new to the project and open source, in general, so I am a little bit lost!